modules/hal_nordic/nrfx/Kconfig

```diff
@@ -573,6 +573,26 @@ config NRFX_UARTE3
 	depends on $(dt_nodelabel_has_compat,uart3,$(DT_COMPAT_NORDIC_NRF_UARTE))
 	select NRFX_UARTE
 
+config NRFX_UARTE_CONFIG_SKIP_GPIO_CONFIG
+	bool "UARTE GPIO configuration support"
+	default y
+
+config NRFX_UARTE_CONFIG_SKIP_PSEL_CONFIG
+	bool "UARTE PSEL configuration support"
+	default y
```

Comment:
why is this required? doesn;t driver use pinctrl unconditionally?

Reply:
we want to tell nrfx to not compile the support for handling GPIO and PSEL in the driver. I changed to so that uart shim is actually implying those options (and removed defaults from here).